AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProviderResponse.swift

```diff
@@ -66,6 +68,33 @@ extension RealtimeConnectionProviderResponse: Decodable {
 /// https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html#error-message
 extension RealtimeConnectionProviderResponse {
 
+    func toConnectionProviderError(connectionState: ConnectionState) -> ConnectionProviderError {
+        guard connectionState != .inProgress else {
+            if isUnauthorizationError() {
+                return .other(errorDescription: "Unauthorized", error: nil, payload: payload)
+            } else {
+                return .connection
+            }
+        }
+
+        if isLimitExceededError() || isMaxSubscriptionReachedError() {
+            // Is it observed that LimitExceeded error does not have `id` while MaxSubscriptionReached does have a
+            // corresponding identifier. Both are mapped to `limitExceeded` with optional identifier.
+            return ConnectionProviderError.limitExceeded(id)
+        } else if isUnauthorizationError() {
+            return .other(errorDescription: "Unauthorized", error: nil, payload: payload)
+        }
```

Comment:
it seems like we check if is an unauthorized error both when the connection is in a `.inProgress` status and when it's not. Should we instead guard on that condition and return early at the beginning of this method?
```swift
guard !isUnauthorizedError() else {
  return .other(errorDescription: "Unauthorized", error: nil, payload: payload)
}

if isLimitExceededError() || isMaxSubscriptionReachedError() { ... } 
```

Reply:
thanks, refactored the code as suggested